### PR TITLE
Correct faulty protocol mapping on oauth2 resource server and add a test

### PR DESCRIPTION
**Pull request by VadimSchmitz**
https://www.pivotaltracker.com/story/show/180861173

### CODE REVIEW

File: tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php [hunk 1]
```diff
@@ -117,4 +118,36 @@ public function provideInvalidProtocols()
...
+    public function test_it_maps_enitty_types_correctly_with_excisting_protocols($inputProtocol, $expectedProtocol)
+    {
+        $this->mockHandler = new MockHandler();
+        $client = m::mock(HttpClientInterface::class);
+        $this->generator = m::mock(GeneratorInterface::class);
+        $this->logger = m::mock(LoggerInterface::class);
+        $this->client = new DeleteManageEntityClient(
+            $client,
+            $this->logger
+        );
```
Comment (MKodde) [inline tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:134]: I copy pasted this from the setUp method during our peer programming session. Now that I'm looking at this code more closely I see you/we can leave out several of the field declarations (generator, logger, mockHandler. They can all be removed from this scope.

File: tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php [hunk 2]
```diff
@@ -117,4 +118,36 @@ public function provideInvalidProtocols()
         yield ['oauth_ccc'];
         yield ['i-dont-exist'];
     }
+
+    /**
+     * @dataProvider provideExpectedProtocolMappings
+     */
+    public function test_it_maps_enitty_types_correctly_with_excisting_protocols($inputProtocol, $expectedProtocol)
```
Comment (MKodde) [inline tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:125]: Typo: ~excisting~ existing

**Review by MKodde — CHANGES_REQUESTED** [review]
Great bugfix!

1. Please be aware of using 'Fix' for every bugfix action you perform in a commit. Naming things is hard, I know. In this instance naming the commit something like: `Correct misconfigured Manage Entity protocol mapping` would be far more descriptive IMO.
2. The other two commits can be squashed into the other two. Thats atomic enough for me. But feel free to keep it a 3 committer :)
3. See some feedback in the code review below.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/Manage/Client/DeleteManageEntityClient.php [hunk 3]
```diff
@@ -82,10 +82,17 @@ public function delete($manageId, $protocol)
         }
     }
 
+    /**
+     * Creates a link between the internal protocol and the protocol defined in manage
+     *
+     * @param $dashboardProtocol
+     * @return mixed
+     * @throws RuntimeException
```
[the remark this answers is not in the document]
Comment (VadimSchmitz) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/Manage/Client/DeleteManageEntityClient.php:90]: Added the param type and cleaned up the PHPdoc

File: tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php (hunk 1 above)
Comment (VadimSchmitz) [reply tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:134]: Cleaned up the test case

File: tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php (hunk 2 above)
Comment (VadimSchmitz) [reply tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:125]: nicely spotted

File: tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php [hunk 4]
```diff
@@ -117,4 +118,33 @@ public function provideInvalidProtocols()
...
+
+        $manageEntityId = 'db2e5c63-3c54-4962-bf4a-d6ced1e9cf33';
+        $client
+            ->shouldReceive('delete')
+            ->with(sprintf('/manage/api/internal/metadata/%s/%s', $expectedProtocol, $manageEntityId))
+            ->once()
+            ->andReturn(true);
+
+        $this->client->delete($manageEntityId, $inputProtocol);
+    }
```
Comment (MKodde) [inline tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:141]: To make the test even more betterest, consider using the `MockeryPHPUnitIntegration` in this trait. 

https://docs.mockery.io/en/latest/reference/phpunit_integration.html

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/Manage/Client/DeleteManageEntityClient.php [hunk 5]
```diff
@@ -82,10 +82,15 @@ public function delete($manageId, $protocol)
         }
     }
 
-    private function getProtocol($dashboardProtocol)
+    /**
+     * Creates a link between the internal protocol and the protocol defined in manage
+     *
+     * @throws RuntimeException
+     */
+    private function getProtocol(String $dashboardProtocol)
```
[the remark this answers is not in the document]
Comment (VadimSchmitz) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/Manage/Client/DeleteManageEntityClient.php:90]: String is now lower cased and returns a string, Thanks for the great explanation in person :)

File: tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php (hunk 4 above)
Comment (VadimSchmitz) [reply tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:141]: it was already extending the class :p
https://github.com/SURFnet/sp-dashboard/blob/741ee78c8d51024a10ca008411f896b408a1ccb5/tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php#L39

Comment (MKodde) [reply tests/unit/Infrastructure/Manage/Client/DeleteEntityClientTest.php:141]: Completely missed that!

**Review by MKodde — APPROVED** [review]
Shipit!

**Pull request merged**